File: src/main/java/com/alealogic/singleproxy/controller/RegistrationController.java
package com.alealogic.singleproxy.controller;

import com.alealogic.singleproxy.model.KeyRequest;
import com.alealogic.singleproxy.service.RegistrationService;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class RegistrationController {

    private RegistrationService registrationService;

    public RegistrationController(RegistrationService registrationService) {
        this.registrationService = registrationService;
    }

    @PostMapping("register-desktop-client")
    public void registerDesktopClient(@RequestBody KeyRequest keyRequest) {
        registrationService.registerDesktopClient(keyRequest.getKey());
    }
}

File: src/main/java/com/alealogic/singleproxy/controller/RegistrationController.kt
package com.alealogic.singleproxy.controller

import org.springframework.web.bind.annotation.RestController
import com.alealogic.singleproxy.service.RegistrationService
import org.springframework.web.bind.annotation.PostMapping
import com.alealogic.singleproxy.model.KeyRequest
import org.springframework.web.bind.annotation.RequestBody

@RestController
class RegistrationController(private val registrationService: RegistrationService) {

    @PostMapping("register-desktop-client")
    fun registerDesktopClient(@RequestBody keyRequest: KeyRequest) {
        registrationService.registerDesktopClient(keyRequest.key)
    }
}